docs/settings.rst

```diff
@@ -58,40 +73,86 @@ Porcupine has these options by default (but you can add your own options too, se
 .. autoclass:: LineEnding
 
 
+Getting and Setting
+-------------------
+
+Use these methods on ``global_settings`` (or other :class:`Settings` objects) for setting and getting values:
+
+.. automethod:: Settings.set
+
+.. method:: Settings.get(option_name, type)
+
+    Return the current value of an option.
+    The *type* should be e.g. ``str`` or ``int`` depending on what type the option is.
+    You can also specify ``object`` to allow any type.
+
+    I have tried to tell mypy that this method returns a value
+    of the given *type*.
+    Mypy understands that in many cases, but not always::
+
+        foo = settings.get('foo', str)
+        reveal_type(foo)  # str
+
+        from pathlib import Path
+        shitty_bar = settings.get('bar', Optional[Path])
+        reveal_type(shitty_bar)  # Any
+
+    Use a type annotation to work around this (and make sure to write the
+    same type two times)::
+
+        good_bar: Path | None = settings.get('bar', Optional[Path])
+        reveal_type(good_bar)  # Optional[Path]
+
+    Before Python 3.10, you can't use the new ``|`` syntax as an argument to ``settings.get()``,
+    even though it otherwise works with ``from __future__ import annotations``.
+    The same goes for built-in generics,
+    such as ``list[str]`` with lower-case ``list``.
+
+    Options of mutable types are returned as copies, so things like
+    ``settings.get('something', List[str])`` always return a new list.
+    If you want to change a setting like that, you need to first get a copy
+    of the current value, then modify the copy, and finally :meth:`set` it
+    back. This is an easy way to make sure that change events run every
+    time the value changes.
+
+.. automethod:: Settings.debug_dump
+
+
 Custom Options
 --------------
 
-You can use :mod:`porcupine.settings` for storing things that shouldn't reset every
+Use :data:`global_settings` to store things that shouldn't reset every
 time Porcupine is restarted.
 
-.. autofunction:: add_option
+.. automethod:: Settings.add_option
 
 Example::
```

Comment:
Example should probably be after a detailed description of the `add_option` method.